### PR TITLE
exec plugin: dynamically allocate grname buffer for getgrnam_r

### PR DESCRIPTION
**Pull request by sreedi**
Dynamically allocate grname buffer in increments of `_SC_GETGR_R_SIZE_MAX/_SC_PAGESIZE`

Fixes: #2696

### CODE REVIEW

File: src/exec.c [hunk 1]
```diff
@@ -408,21 +414,65 @@ static int fork_child(program_list_t *pl, int *fd_in, int *fd_out,
...
         grbuf_size = 4096;
-      char grbuf[grbuf_size];
 
-      status = getgrnam_r(pl->group, &gr, grbuf, sizeof(grbuf), &gr_ptr);
-      if (status != 0) {
-        ERROR("exec plugin: Failed to get group information "
-              "for group ``%s'': %s",
-              pl->group, STRERROR(status));
-        goto failed;
+      long int size;
```
Comment (octo) [inline src/exec.c]: The code doing the group name to ID lookup is getting quite long and blowing up the `fork_child()` function. Could you please move that code (essentially everything in this block) to its own function?

Comment (sreedi) [reply src/exec.c]: Hi @octo , 
I have made the change to refactor this block of group ID lookup code into a separate function. Please review.

**Comment by octo** [comment]
Awesome, thank you so much @sreedi! :tada:

**Comment by sreedi** [comment]
Thanks @octo, for your quick response and support.

**Pull request merged**